DataStructures/DoublyLinkedList/DoublyLinkedList.cs

```diff
@@ -0,0 +1,339 @@
+            while (current != null)
+            {
+                yield return current.Data;
+                current = current.Next;
+            }
+        }
+
+        /// <summary>
+        /// Gets an enumerable based on the data in the list reversed.
+        /// </summary>
+        /// <returns>The data in the list in an IEnumerable. It can used to create a list or an array with LINQ.</returns>
+        public IEnumerable<T> GetDataReversed()
+        {
+            var current = Tail;
+            while (current != null)
+            {
+                yield return current.Data;
+                current = current.Previous;
+            }
+        }
+
+        /// <summary>
+        /// Reverses the list. Because of how doubly linked list are structured this is not a complex action.
+        /// </summary>
+        public void Reverse()
+        {
+            DoublyLinkedListNode<T> current = Head;
+            DoublyLinkedListNode<T> temp = null;
+
+            while (current != null)
+            {
+                temp = current.Previous;
+                current.Previous = current.Next;
+                current.Next = temp;
+                current = current.Previous;
+            }
+
+            Tail = Head;
+
+            // temp can be null on empty list
+            if (temp != null)
+            {
+                Head = temp.Previous;
+            }
+        }
+
+        /// <summary>
+        /// Looks for a node in the list that contains the value of the parameter.
+        /// </summary>
+        /// <param name="data"> Value to be looked for in a node.</param>
+        /// <returns>The node in the list the has the paramater as a value or null if not found.</returns>
+        public DoublyLinkedListNode<T> Find(T data)
+        {
+            var current = Head;
+            while (current != null)
+            {
+                if (current.Data.Equals(data))
+                {
+                    return current;
+                }
+
+                current = current.Next;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Looks for a node in the list that contains the value of the parameter.
+        /// </summary>
+        /// <param name="position"> Position in the list.</param>
+        /// <returns>The node in the list the has the paramater as a value or null if not found.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when position is negative or out range of the list.</exception>
+        public DoublyLinkedListNode<T> GetAt(int position)
+        {
+            if (position < 0 || position >= Count)
+            {
+                throw new ArgumentOutOfRangeException($"Max count is {Count}");
+            }
+
+            var current = Head;
+            for (var i = 0; i < position; i++)
+            {
+                if (current is null)
+                {
+                    throw new NullReferenceException("current");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['System.NullReferenceException' should not be thrown by user code.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4803417)